src/main/java/org/jboss/forge/arquillian/command/algeron/AlgeronProviderValidateContractCommand.java

```diff
@@ -0,0 +1,139 @@
+import org.jboss.forge.addon.parser.java.resources.JavaResource;
+import org.jboss.forge.addon.parser.java.resources.JavaResourceVisitor;
+import org.jboss.forge.addon.projects.Project;
+import org.jboss.forge.addon.projects.ProjectFactory;
+import org.jboss.forge.addon.projects.ui.AbstractProjectCommand;
+import org.jboss.forge.addon.resource.visit.VisitContext;
+import org.jboss.forge.addon.ui.command.UICommand;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.input.UIInput;
+import org.jboss.forge.addon.ui.input.UISelectOne;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.metadata.WithAttributes;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.arquillian.api.algeron.AlgeronSetupFacet;
+import org.jboss.forge.arquillian.container.model.ContractConsumerLibrary;
+import org.jboss.forge.arquillian.container.model.ContractProviderLibrary;
+import org.jboss.forge.arquillian.testframework.algeron.AlgeronProvider;
+import org.jboss.forge.roaster.model.JavaType;
+import org.jboss.forge.roaster.model.source.JavaClassSource;
+
+import javax.inject.Inject;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class AlgeronProviderValidateContractCommand extends AbstractProjectCommand implements UICommand {
+
+    @Inject
+    private ProjectFactory projectFactory;
+
+    @Inject
+    @WithAttributes(shortName = 'j', label = "Test Class", required = true)
+    private UISelectOne<JavaClassSource> testClass;
+
+    @Inject
+    @WithAttributes(shortName = 'p', label = "Provider Name", required = true)
+    private UIInput<String> provider;
+
+    @Inject
+    @WithAttributes(shortName = 't', label = "Contract Type", required = false, enabled = false)
+    private UISelectOne<ContractProviderLibrary> contractType;
+
+    @Override
+    public UICommandMetadata getMetadata(UIContext context) {
+        return Metadata.from(super.getMetadata(context), getClass())
+            .category(Categories.create("Algeron"))
+            .name("Arquillian Algeron: Create Provider Test")
+            .description("This command creates skeleton for validating provider on given test");
+    }
+
+    @Override
+    public void initializeUI(UIBuilder builder) throws Exception {
+        builder.add(contractType)
+            .add(testClass)
+            .add(provider);
+
+        final Project project = getSelectedProject(builder);
+
+        final List<JavaClassSource> sources = new ArrayList<>();
+        project.getFacet(JavaSourceFacet.class).visitJavaTestSources(new JavaResourceVisitor() {
```

Comment:
![dr -mccoy-and-captain-kirk-agree-on-star-trek](https://cloud.githubusercontent.com/assets/719616/22938555/6409c600-f2dc-11e6-8c75-7d273f24633f.gif)